### PR TITLE
Add proxied canister call spec

### PR DESCRIPTION
**Pull request by frederikrothenberger**
This PR adds a possible specification for proxied canister calls. It is intended to be implemented by IDPs that provide a front-end for end users. The goal is to have a generic specification that works for existing canister calls (given the canister can be extended to incorporate the consent canister interface).

[Rendered version of the specification](https://github.com/dfinity/wg-identity-authentication/blob/frederik/proxy_call/topics/proxied_canister_calls.md)

Possible next steps:
  * Discuss with working group
  * Security review
  * Add concrete end to end example

### CODE REVIEW

**Review by Dfinity-Bjoern — COMMENTED** [review]
I see how this works on a technical level. I assume the intention is to use it for infrequent actions?

**Comment by frederikrothenberger** [comment]
> I see how this works on a technical level. I assume the intention is to use it for infrequent actions?

Yes, exactly. I.e. asset transfers and other high value operations, where you would want user consent anyways.

File: topics/proxied_canister_calls.md [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+This specification describes a protocol for proxying canister calls through an identity provider (IDP) on the IC. This gives dapps on the IC the ability to send canister calls to other services on the users behalf if explicitly consented to by the end-user.
+
+A common application of such a protocol would be transaction approval: I.e. send a transfer request to the service that holds the users assets.
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
+"SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this
+document are to be interpreted as described in [RFC2119](https://datatracker.ietf.org/doc/html/rfc2119).
+
+## Assumptions
+* The browser and the IDP can be trusted.
```
[the remark this answers is not in the document]
Comment (frederikrothenberger) [reply topics/proxied_canister_calls.md]: Yes, I have made this more clear.

File: topics/proxied_canister_calls.md [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+service : {
+    validate_proxy_call: (proxy_call_validation) -> (proxy_call_validation_result);
+}
+```
+
+## Validation of Derivation Origin
+
+If the IDP uses principal derivation based on the front-end origin, the target canister must be validated against that origin. This validation prevents the derived principal from being used in unintended interactions or being leaked to malicious canisters.
+
+To validate a target canister against an origin, the asset `/.well-known/ic-proxy-config` must be fetched from that origin. The target canister is allowed to be called with a principal from this origin if one of the following conditions is true:
```
[the remark this answers is not in the document]
Comment (frederikrothenberger) [reply topics/proxied_canister_calls.md:126]: Thanks, fixed.

**Comment by neeboo** [comment]
There is one situation that is needed to consider, the `allowTargetCanisters` can be dynamic. When new canisters are pushed to this array. Will the authentication process start again or start from somewhere in the middle?

**Comment by frederikrothenberger** [comment]
> There is one situation that is needed to consider, the `allowedTargetCanisters` can be dynamic. When new canisters are pushed to this array. Will the authentication process start again or start from somewhere in the middle?

The proxy calling flow needs to be repeated for each a call that is made. This means that for every proxy call, the `allowedTargetCanisters` would be fetched new.
--> Changes to `allowedTargetCanisters` will affect all proxy call flows that were started after the change was made (which I think is fine from a security perspective).

**Comment by frederikrothenberger** [comment]
Superseded by ICRC-21 and ICRC-25.

**Pull request closed without merging**